### PR TITLE
Photo Sharing sample app with Auth, Storage and AppSync.

### PR DESCRIPTION
**Pull request by changxu0306**
Description of this PR: The whole sample app code is committed to this PR, with Auth, AppSync and Storage implemented.
* Define Album class and Photo class for album and photo objects. Make it clear for Object-Oriented Programming.
* AppSyncHelper class provides AppSync utilities for the sample app. When an AppSyncHelper object is created, it will initialize AWS AppSyncClient with Cognito User Pools by its constructor. Within the class, we provide multiple queries and mutations (create, update, delete) for albums and photos.
* StorageHelper class provides Storage utilities (TransferUtility) for the sample app. It also provides useful type convertion APIs between bitmaps, files, drawables and URIs.
* AlbumActivity view displays all the albums belong to the user who signed in. User can create an album, delete an album, update an album's name or sign out in this activity.
* PhotoActivity view allows user upload an image file and download image files from Amazon S3 bucket. After download each image file, it will show up on the screen in a grid view.
* DownloadSelectionActivity view shows a list of items which are stored in the S3 bucket. User can choose one file at a time to download by clicking on an item of the list. After clicking, it will go back to PhotoActivity.
* AlbumAdapter and PhotoAdapter are customized adapters which bind attributes of an Album or Photo object with a UI item. Listeners are defined when clicking add button and delete button. It allows UI changes dynamically with the latest view.

### CODE REVIEW

File: CircleciScripts/test_names.json [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
+	"tests" : [
+		{
+			"auth" : "AuthUITest",
+			"api_1" : "AppSyncMutationUITest",
```
Comment (kvasukib) [inline CircleciScripts/test_names.json]: make api as the category name with the tests in an array.

File: PhotoAlbumSample/.gitignore [hunk 2]
```diff
@@ -0,0 +1,5 @@
+# amplify codegen
```
Comment (kvasukib) [inline PhotoAlbumSample/.gitignore]: add awsconfiguration.json in this file.

File: PhotoAlbumSample/app/build.gradle [hunk 3]
```diff
@@ -10,6 +10,11 @@ android {
         versionCode 1
         versionName "1.0"
         testInstrumentationRunner "android.support.test.runner.AndroidJUnitRunner"
+        // The following argument makes the Android Test Orchestrator run its
+        // "pm clear" command after each test invocation. This command ensures
+        // that the app's state is completely cleared between tests.
+//        testInstrumentationRunnerArguments clearPackageData: 'true'
```
Comment (kvasukib) [inline PhotoAlbumSample/app/build.gradle]: Why is this commented?

File: CircleciScripts/test_names.json (hunk 1 above)
Comment (changxu0306) [reply CircleciScripts/test_names.json]: change the type of values of all categories to array:
{
	"tests" : [
		{
			"auth" : ["AuthUITest"],
			"api" : ["AppSyncMutationUITest", "AppSyncQueryUITest"],
			"storage" : ["StorageUITest"]
		}
	]
}

**Pull request merged**